### PR TITLE
rng: use PathBuf for socket path

### PR DESCRIPTION
**Pull request by epilys**
### Summary of the PR

clap can parse a PathBuf directly from the command line arguments, and paths are not always UTF-8. Use PathBuf instead of a String to allow for all valid paths.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Comment by epilys** [comment]
CI on staging failed because of network error (it could not connect to the pipewire git)

File: vhost-device-rng/src/main.rs [hunk 1]
```diff
@@ -57,58 +58,82 @@ struct RngArgs {
...
+            .file_name()
+            .expect("socket_path has no filename.");
+        let socket_file_parent = self
+            .socket_path
+            .parent()
+            .expect("socket_path has no parent directory.");
+
+        let make_socket_path = |i: u32| -> PathBuf {
+            let mut file_name = socket_file_name.to_os_string();
+            file_name.push(std::ffi::OsStr::new(&i.to_string()));
```
Comment (Ablu) [inline vhost-device-rng/src/main.rs:125]: OsStr::from(i.to_string()) probably does the conversion in "zero-copy"

**Review by Ablu — APPROVED** [review]

File: vhost-device-rng/src/main.rs (hunk 1 above)
Comment (epilys) [reply vhost-device-rng/src/main.rs:125]: OsStr is a borrowed ref, so there's no copying. Do you mean something else?

Also OsStr doesn't implement From<String> [from what I see here](https://doc.rust-lang.org/std/ffi/struct.OsStr.html)...

Comment (Ablu) [reply vhost-device-rng/src/main.rs:125]: Ah. I read it as OsString... Confusing that str is lower-case but OsStr is camel... Looks good.
Still need to get out of the C++ habit of being afraid of seeing `&` and getting paranoid of whether reference to temporary objects stay alive...

**Review by stefano-garzarella — APPROVED** [review]

**Pull request merged**